core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -425,10 +425,9 @@ private[spark] class TaskSchedulerImpl(
           }
         }
       } else {
-        // No task sets are active but we still got an error. Just exit since this
-        // must mean the error is during registration.
-        // It might be good to do something smarter here in the future.
-        throw new SparkException(s"Exiting due to error from cluster scheduler: $message")
+        // No task sets are active but we still got an error. Just log the error
+        // since this must mean the error is during registration.
+        logError(s"Exiting due to error from cluster scheduler: $message")
```

Comment:
that means, for Errors, like `VirtualMachineError`,  you may want to terminate the process, but for others like SparkException, it just stop the SparkContext()